infrastructure/keyvault.tf

```diff
@@ -7,5 +7,5 @@ module "azurekeyvault" {
   product_group_name      = "DTS Platform Operations"
   common_tags             = local.common_tags
   create_managed_identity = true
-
-}
\ No newline at end of file
+  object_id               = data.azurerm_client_config.current.object_id
```

Comment:
I had to partially revert the cnp-module-key-vault PR because when using a managed identity the object_id is empty string, we could enhance it to use client_config if it's a non empty string but let's see how often it's a pain